include/Framework/Performance/Tracker.h

```diff
@@ -0,0 +1,68 @@
+#ifndef FRAMEWORK_PERFORMANCE_TRACKER
+#define FRAMEWORK_PERFORMANCE_TRACKER
+
+#include <map>
+
+#include <TTree.h>
+#include <TDirectory.h>
+
+#include "Framework/Performance/Timer.h"
+#include "Framework/Performance/Callback.h"
+
+namespace framework::performance {
+
+/**
+ * Class to interface between framework::Process and various measurements
+ * that can eventually be written into the output histogram file.
+ *
+ * @see Timer for the data format of timing measurements
+ */
+class Tracker {
+ public:
+  /**
+   * Create the tracker with a specific destination for writing information
+   *
+   * @param[in] storage_directory directory in-which to write data when closing
+   * @param[in] names sequence of processor names we will be tracking
+   */
+  Tracker(TDirectory *storage_directory, const std::vector<std::string>& names);
+  /**
+   * Close up tracking and write all of the data collected to the storage directory
+   */
+  ~Tracker();
+  /// literally first line of Process::run
+  void absolute_start();
+  /// literally last line of Process::run (if run compeletes without error)
+  void absolute_end();
+  /// start the timer for a specific callback and specific processor
+  void start(Callback cb, std::size_t i_proc);
+  /// end the timer for a specific callback and specific processor
+  void end(Callback cb, std::size_t i_proc);
```

Comment:
I've made the executive decision to be lazy :monocle_face: 

I don't think it will have any effect that is distinguishable even if we have optimizations off. If we have optimizations on, I'm fairly certain the compiler will be able to deduce the same assembly without requiring us to do fancy footwork with `template` in order to move `cb` to a compile-time argument.

Reply:
I made the `to_index` function `constexpr` but I'm not really sure where to go from here - feel free to try something yourself if you have an idea? I'm not as well-read on `constexpr` as you I believe.